src/main/java/link/infra/indium/renderer/render/TerrainRenderContext.java

```diff
@@ -21,16 +22,23 @@
 import java.util.function.Function;
 
 public class TerrainRenderContext extends AbstractRenderContext {
-	private final TerrainBlockRenderInfo blockInfo = new TerrainBlockRenderInfo();
+	private final TerrainBlockRenderInfo blockInfo;
 	private final ChunkRenderInfo chunkInfo = new ChunkRenderInfo();
-	private final AoCalculator aoCalc = new AoCalculator(blockInfo, chunkInfo::cachedBrightness, chunkInfo::cachedAoLevel);
+	private final AoCalculator aoCalc;
 
 	private Vec3i origin;
 	private Vec3d modelOffset;
 
-	private final BaseMeshConsumer meshConsumer = new BaseMeshConsumer(new QuadBufferer(chunkInfo::getChunkModelBuilder), blockInfo, aoCalc, this::transform);
+	private final BaseMeshConsumer meshConsumer;
 
-	private final BaseFallbackConsumer fallbackConsumer = new BaseFallbackConsumer(new QuadBufferer(chunkInfo::getChunkModelBuilder), blockInfo, aoCalc, this::transform);
+	private final BaseFallbackConsumer fallbackConsumer;
+
+	public TerrainRenderContext(BlockOcclusionCache blockOcclusionCache) {
```

Comment:
Could you revert the rest of the changes in this class and move the `BlockOcclusionCache` parameter to `prepare`? Then it can be set in `MixinChunkRenderRebuildTask` with the rest of the rendering context parameters.